int32[] ids  # desc to be updated
---
int32[] ids   # affected instances
